SMLHelper/Patchers/PrefabDatabasePatcher.cs

```diff
@@ -1,82 +1,134 @@
+            [PatchUtils.Postfix]
+            [HarmonyPatch(typeof(PrefabDatabase), nameof(PrefabDatabase.LoadPrefabDatabase))]
+            internal static void LoadPrefabDatabase_Postfix()
             {
-                PrefabDatabase.prefabFiles[prefab.ClassID] = prefab.PrefabFileName;
+                foreach (ModPrefab prefab in ModPrefab.Prefabs)
+                {
+                    PrefabDatabase.prefabFiles[prefab.ClassID] = prefab.PrefabFileName;
+                }
+
+                var tryGetPrefabFilename = AccessTools.Method(typeof(PrefabDatabase), nameof(PrefabDatabase.TryGetPrefabFilename));
+                Initializer.harmony.Unpatch(tryGetPrefabFilename, HarmonyPatchType.Prefix, Initializer.harmony.Id);
             }
+        }
 
-            Initializer.harmony.Unpatch(tryGetPrefabFilename, HarmonyPatchType.Prefix, Initializer.harmony.Id);
+        [PatchUtils.Prefix]
+        [HarmonyPatch(typeof(PrefabDatabase), nameof(PrefabDatabase.TryGetPrefabFilename))]
+        internal static bool TryGetPrefabFilename_Prefix(string classId, ref string filename, ref bool __result)
+        {
+            if (!ModPrefab.TryGetFromClassId(classId, out ModPrefab prefab))
+                return true;
+
+            filename = prefab.PrefabFileName;
+            __result = true;
+            return false;
         }
 
+        [PatchUtils.Prefix] // SUBNAUTICA_EXP TODO: remove for SN after async update
+        [HarmonyPatch(typeof(PrefabDatabase), "GetPrefabForFilename")] // method can be absent
         internal static bool GetPrefabForFilename_Prefix(string filename, ref GameObject __result)
         {
-            if (ModPrefab.TryGetFromFileName(filename, out ModPrefab prefab))
-            {
-                GameObject go = prefab.GetGameObjectInternal();
-                __result = go;
-
-                return false;
-            }
+            if (!ModPrefab.TryGetFromFileName(filename, out ModPrefab prefab))
+                return true;
 
-            return true;
+            __result = prefab.GetGameObjectInternal();
+            return false;
         }
 
-        internal static bool TryGetPrefabFilename_Prefix(string classId, ref string filename, ref bool __result)
+        private static IPrefabRequest GetModPrefabAsync(string classId)
         {
-            if (ModPrefab.TryGetFromClassId(classId, out ModPrefab prefab))
+            if (!ModPrefab.TryGetFromClassId(classId, out ModPrefab prefab))
+                return null;
+
+            try
             {
-                filename = prefab.PrefabFileName;
-                __result = true;
-                return false;
+                // trying sync method for backward compatibility
+                if (prefab.GetGameObjectInternal() is GameObject go) // SUBNAUTICA_EXP TODO: remove for SN after async update
+                    return new LoadedPrefabRequest(go);
             }
-            return true;
+            catch (Exception) {}
```

Comment:
Empty catch block?  
Shouldn't we be at least logging something here?

Reply:
It's for the time after async update, when sync methods will not work. Before update that catch is not supposed to catch anything, and after update there will be probably lot of exceptions here. And after exception this patch will try async request.
Probably not a perfect solution and this block is supposed to be removed after async update anyway.